core-bundle/src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -348,32 +348,55 @@ private function setCacheHeaders(Response $response)
 		/** @var $objPage PageModel */
 		global $objPage;
 
+		// Do not cache response if caching was not configured at all and make sure the response is private.
 		if (($objPage->cache === false || $objPage->cache < 1) && ($objPage->clientCache === false || $objPage->clientCache < 1))
 		{
-			$response->headers->addCacheControlDirective('no-cache');
-			$response->headers->addCacheControlDirective('no-store');
+			$response->headers->set('Cache-Control', 'no-cache, no-store');
```

Comment:
Is this any different from `addCacheControlDirective()`?

Reply:
It's a) shorter and b) it overrides any existing definition which is good.